ccan/ccan/tal/path/path.c

```diff
@@ -321,8 +321,17 @@ char *path_rel(const tal_t *ctx, const char *from, const char *to)
 			goto fail;
 	}
 
+#pragma GCC diagnostic push
```

Comment:
ccan is an upstream library: http://ccodearchive.net/

So we should push any updates to ccan https://github.com/rustyrussell/ccan/

Reply:
Hi,
Thank you giving me ccan info. I'll try to update ccan too!
I don't like cluttered code too! But the root cause is sharing code both GCC and Clang.
And, I just want to reduce false positive warnings to raise more important warnings.
I don't mind the actual method in this case (initialize variable, add many preprocessor directives, or, change Makefile...). Anyway, Lightning users don't see it finally.
I hope the reviewers' policies are in line.

Reply:
Hi ZmmSCPxj,
I pushed an updated to cpan https://github.com/rustyrussell/ccan  [Suppress false positive warning "-Wmaybe-uninitialized" with GCC -O3]: https://github.com/rustyrussell/ccan/pull/83
I'm in the status waiting for any reply. 
BR,